summary/summary.go

```diff
@@ -68,11 +68,21 @@ func (s *StatsSummary) AddSample(stat info.ContainerStats) error {
 	s.secondSamples = append(s.secondSamples, &sample)
 	s.updateLatestUsage()
 	// TODO(jnagal): Use 'available' to avoid unnecessary computation.
-	if len(s.secondSamples) == 60 {
-		// Make a minute stat.
+	numSamples := len(s.secondSamples)
+	elapsed := int64(0)
+	if numSamples > 1 {
+		start := s.secondSamples[0].Timestamp
+		end := s.secondSamples[numSamples-1].Timestamp
+		elapsed = end.Sub(start).Nanoseconds()
+	}
+	if elapsed > 60*time.Second.Nanoseconds() {
```

Comment:
Should we worry if this is >> 60s, for example if it is 120s we should in theory keep the stats over the 60s. I'm guessing if we add one stat at a time this is not a problem though. Is that what we expect?